File: readability/readability2.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

int count_letters(string s);

int main(void)
{
    string s = get_string("Text: ");
    int l = count_letters(s);
    printf("%i", l);
    printf(" letters\n");
}

int count_letters(string s)
{
    int i = 0;
    for (int j = 0, n = strlen(s); j < n; j++)
    {
        while (isalpha(s[i]))
        {
            return i++;
        }
    }
    return i;
}

File: sentimental-readability/readability2.py
from cs50 import get_string

text = get_string("Text: ")

# count sentences
se = text.count(".") + text.count("!") + text.count("?")

# count words
w = text.count(" ") + 1

# count letters
l = 0
for i in range(len(text)):
    if text[i].isalpha():
        +l

L = round(l/w * 100)
S = (se/w * 100)
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")